lms/djangoapps/mobile_api/testutils.py

```diff
@@ -123,8 +146,228 @@ def test_other_user(self):
         self.api_response(expected_response_code=404, username=other.username)
 
 
+class MobileAPIMilestonesMixin(object):
+    """
+    Tests the Mobile API decorators for milestones.
+
+    The two milestones supported in these tests are entrance exams and
+    pre-requisite courses. If either of these milestones are unfulfilled,
+    the mobile api will appropriately block content until the milestone is
+    fulfilled.
+    """
+    MILESTONE_ERROR = {'developer_message': 'Cannot access content with unfulfilled pre-requisites or unpassed entrance exam.'}  # pylint: disable=line-too-long
+
+    # Enrollment list hides enrolled courses, milestone courses are not hidden
+    ALLOW_ACCESS_TO_MILESTONE_COURSE = False  # pylint: disable=invalid-name
+
+    def __init__(self):
+        self.entrance_exam = None
+        self.problem_1 = None
+        self.milestone = None
+        self.milestone_relationship_types = None
+        self.request = None
+
+    def init_course_access(self, course_id=None):
+        """Base implementation of initializing the user for each test."""
+        self.login_and_enroll(course_id)
+
+    def _add_entrance_exam(self):
+        """ Sets up entrance exam """
+        try:
+            # in case milestone relationship types have already been set
+            seed_milestone_relationship_types()
+        except IntegrityError:
+            pass
+
+        # Set up the extrance exam
+        self.course.entrance_exam_enabled = True
+
+        self.entrance_exam = ItemFactory.create(
+            parent=self.course,
+            category="chapter",
+            display_name="Entrance Exam Chapter",
+            is_entrance_exam=True,
+            in_entrance_exam=True
+        )
+        self.problem_1 = ItemFactory.create(
+            parent=self.entrance_exam,
+            category='problem',
+            display_name="The Only Exam Problem",
+            graded=True,
+            in_entrance_exam=True
+        )
+
+        namespace_choices = get_namespace_choices()
+        milestone_namespace = generate_milestone_namespace(
+            namespace_choices.get('ENTRANCE_EXAM'),
+            self.course.id
+        )
+        self.milestone = {
+            'name': 'Test Milestone',
+            'namespace': milestone_namespace,
+            'description': 'Entrance Exam for TestMobileAPIMilestones',
+        }
+        self.milestone_relationship_types = get_milestone_relationship_types()
+        self.milestone = add_milestone(self.milestone)
+
+        self.course.entrance_exam_minimum_score_pct = 0.50
+        self.course.entrance_exam_id = unicode(self.entrance_exam.scope_ids.usage_id)
```

Comment:
`self.entrance_exam.scope_ids.usage_id` can just be `self.entrance_exam.location`.